sdk/openai/openai/test/public/utils/asserts.ts

```diff
@@ -0,0 +1,469 @@
+  assert.isString(choice.text);
+  ifDefined((choice as any).content_filter_results, assertContentFilterResultsForChoice);
+}
+
+function assertContentFilterResultsForChoice(cfr: ContentFilterResultsForChoiceOutput): void {
+  if (cfr.error) {
+    assertContentFilterErrorResults(cfr.error);
+  } else {
+    ifDefined(cfr.hate, assertContentFilterResult);
+    ifDefined(cfr.self_harm, assertContentFilterResult);
+    ifDefined(cfr.sexual, assertContentFilterResult);
+    ifDefined(cfr.violence, assertContentFilterResult);
+    ifDefined(cfr.profanity, assertContentFilterResult);
+    ifDefined(cfr.custom_blocklists, assertContentFilterBlocklistIdResult);
+    ifDefined(cfr.protected_material_code, assertContentFilterCitedDetectionResult);
+    ifDefined(cfr.protected_material_text, assertContentFilterDetectionResult);
+  }
+}
+
+function assertContentFilterResultsForPrompt(cfr: ContentFilterResultsForPromptOutput[]): void {
+  assert.isArray(cfr);
+  for (const item of cfr) {
+    assertContentFilterResultsForPromptItem(item);
+  }
+}
+
+function assertContentFilterCitedDetectionResult(
+  val: ContentFilterCitedDetectionResultOutput,
+): void {
+  assert.isBoolean(val.detected);
+  assert.isBoolean(val.filtered);
+  assert.isString(val.license);
+  ifDefined(val.URL, assert.isString);
+}
+
+function assertContentFilterResultsForPromptItem(cfr: ContentFilterResultsForPromptOutput): void {
+  assert.isNumber(cfr.prompt_index);
+  assertContentFilterResultDetailsForPrompt(cfr.content_filter_results);
+}
+
+function assertContentFilterResultDetailsForPrompt(
+  cfr: ContentFilterResultDetailsForPromptOutput,
+): void {
+  if (cfr.error) {
+    assertContentFilterErrorResults(cfr.error);
+  } else {
+    ifDefined(cfr.hate, assertContentFilterResult);
+    ifDefined(cfr.self_harm, assertContentFilterResult);
+    ifDefined(cfr.sexual, assertContentFilterResult);
+    ifDefined(cfr.violence, assertContentFilterResult);
+    ifDefined(cfr.profanity, assertContentFilterDetectionResult);
+    ifDefined(cfr.jailbreak, assertContentFilterDetectionResult);
+    ifDefined(cfr.custom_blocklists, (arr) =>
+      assertArray(arr, assertContentFilterBlocklistIdResult),
+    );
+  }
+}
+
+function assertContentFilterErrorResults(error: ErrorModel): void {
+  assert.isDefined(error);
+  assert.isDefined(error.code);
+  assert.isDefined(error.message);
+}
+
+function assertContentFilterResult(val: ContentFilterResultOutput): void {
+  assert.isBoolean(val.filtered);
+  assert.isString(val.severity);
+}
+
+function assertContentFilterDetectionResult(val: ContentFilterDetectionResultOutput): void {
+  assert.isBoolean(val.detected);
+  assert.isBoolean(val.filtered);
+}
+
+function assertContentFilterBlocklistIdResult(val: ContentFilterBlocklistIdResultOutput): void {
+  assert.isString(val.id);
+  assert.isBoolean(val.filtered);
+}
+
+function assertChoice(
+  choice:
+    | OpenAI.Chat.Completions.ChatCompletion.Choice
```

Comment:
Why some types are referenced using a qualified name starting from `OpenAI` and some others are imported and referenced directly?